### DISCUSSION TITLE
Can dialogue be triggered without using _unhandled_input?

### DISCUSSION THREAD

**Discussion by danfriedz** [nathanhoad/godot_dialogue_manager#403] [Q&A] [answered]
In my taxi game, i have built a system to activate destinations when the passenger pickup commences. I would prefer to automatically start the dialogue when they appear in the vehicle.

If i try to call 
`DialogueManager.show_dialogue_balloon(character_1_dialogue,"beat_0")`
within the pickup passenger function, i get this error
`Invalid set index 'visible' (on base: 'Nil') with value of type 'bool'.`
this is occurring on this line of code in 'example_balloon.gd'
`var dialogue_line: DialogueLine:`
`    set(next_dialogue_line):`
`   ***misc stuff*****`
`     character_label.visible = not dialogue_line.character.is_empty()`
member variables for dialogue_line are null 

I've read through the docs and couldn't find an answer. Am I implementing this wrong?

Godot version 4.2.stable
The  _unhandled_input method works find btw. So the plugin appears to be functioning.

**Comment by nathanhoad** [comment, marked as answer]
I just pushed something that should fix that. While you're waiting for the next release you can give it a go yourself by [making this change](https://github.com/nathanhoad/godot_dialogue_manager/commit/0e23317c7cf7d0aa953670042d66ce3e6a97f879).

**Reply by danfriedz** [reply]
Thanks, that solves the crash.
I was able to get it working how I wanted.
Ty for providing a fix!

If anyone stumbles upon this I just instantiate a scene that has code a little like this
```
#display the dialogue box instantly upon loading scene
func _ready() -> void:
	DialogueManager.show_dialogue_balloon(dialogue_file,beat)

#click through for the rest of the dialogue
func _unhandled_input(event: InputEvent) -> void:
	if Input.is_action_just_pressed("dialogic_default_action"):
		DialogueManager.show_dialogue_balloon(dialogue_file,beat)
```

Video:
https://github.com/nathanhoad/godot_dialogue_manager/assets/61813510/8bc13a40-071e-4b12-9952-b9d6aebf7dbf
